mlir/tools/mlir-miopen-lib/mlir-miopen-lib.cpp

```diff
@@ -259,10 +259,10 @@ extern "C" MiirStatus miirLowerTuningParams(MiirHandle mlirHandle) {
   PassManager pm(module.getContext(), PassManager::Nesting::Implicit);
 
   // Passes for lowering MIOpen dialect.
-  pm.addPass(mlir::miopen::createLowerMIOpenOpsStep1Pass());
-  pm.addPass(mlir::miopen::createAffineTransformPass());
   pm.addPass(
       mlir::miopen::createAffixTuningParametersPass(0, 0, handle->perfConfig));
+  pm.addPass(mlir::miopen::createLowerMIOpenOpsStep1Pass());
+  pm.addPass(mlir::miopen::createAffineTransformPass());
```

Comment:
Can/should those passes fail? If not, we can remove them from the API. This reduce the overhead for the client to do a quick check.

Reply:
We might need to introduce `miopen-lowering-step2` here. That is the pass which would actually fail in case we pass in unsuitable tuning parameters.